hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/UgiFilter.java

```diff
@@ -0,0 +1,169 @@
+      headerMap.put(headerKey, httpRequest.getHeader(headerKey));
+      LOG.info("request {} : {}", headerKey, httpRequest.getHeader(headerKey));
+    }
+    AWSSignatureProcessor signature = new AWSSignatureProcessor(headerMap,
+        httpRequest.getParameterMap());
+    SignatureInfo signatureInfo;
+    String stringToSign = "";
+    String awsAccessId = "";
+    try {
+      signatureInfo = signature.parseSignature();
+      if (signatureInfo.getVersion() == SignatureInfo.Version.V4) {
+        stringToSign =
+            StringToSignProducer.createSignatureBase(signatureInfo,
+                httpRequest.getScheme(),
+                httpRequest.getMethod(),
+                httpRequest.getPathInfo(),
+                AWSSignatureProcessor.LowerCaseKeyStringMap
+                    .fromHeaderMap(headerMap),
+                StringToSignProducer.fromMultiValueToSingleValueMap(
+                    httpRequest.getParameterMap()));
+      }
+      awsAccessId = signatureInfo.getAwsAccessId();
+      validateAccessId(awsAccessId);
+    } catch (Throwable t) {
+      throw new IOException(t);
+    }
+
+    UserGroupInformation remoteUser =
+        UserGroupInformation.createRemoteUser(awsAccessId);
+
+    if (OzoneSecurityUtil.isSecurityEnabled(ozoneConfiguration)) {
+      LOG.debug("Creating s3 auth info for client.");
+
+      if (signatureInfo.getVersion() == SignatureInfo.Version.NONE) {
+        //throw MALFORMED_HEADER;
+        throw new IOException("MALFORMED_HEADER");
+      }
+
+      OzoneTokenIdentifier identifier = new OzoneTokenIdentifier();
+      identifier.setTokenType(S3AUTHINFO);
+      identifier.setStrToSign(stringToSign);
+      identifier.setSignature(signatureInfo.getSignature());
+      identifier.setAwsAccessId(awsAccessId);
+      identifier.setOwner(new Text(awsAccessId));
+      if (LOG.isTraceEnabled()) {
+        LOG.trace("Adding token for service:{}", omService);
+      }
+      Token<OzoneTokenIdentifier> token = new Token(identifier.getBytes(),
+          identifier.getSignature().getBytes(StandardCharsets.UTF_8),
+          identifier.getKind(),
+          omService);
+      remoteUser.addToken(token);
+    }
+    try {
+      remoteUser.doAs((PrivilegedExceptionAction<Void>) () -> {
+        filterChain.doFilter(httpRequest, servletResponse);
+        return null;
+      });
+    } catch (InterruptedException e) {
+      throw new IOException("Interrupted thread call doAs", e);
+    }
+  }
+
+  @Override
+  public void destroy() { }
+
+  private WebApplicationException wrapOS3Exception(OS3Exception os3Exception) {
+    return new WebApplicationException(os3Exception,
+        os3Exception.getHttpCode());
+  }
+
+  private void validateAccessId(String awsAccessId) throws Exception {
+    if (awsAccessId == null || awsAccessId.equals("")) {
+      LOG.error("Malformed s3 header. awsAccessID: ", awsAccessId);
+      throw wrapOS3Exception(MALFORMED_HEADER);
+    }
+  }
```

Comment:
Is access ID validation going to be refined in a later patch, or is this all that's needed?

Reply:
Thanks @adoroszlai for the comments.  There is no further refinement for validating the access ID.

The s3 client does simple header checking for expected fields required to process s3 request. It is checked  here and with the s3gateway SignatureParser implementation.  On the server side, the id is used for request authentication.